growthbook/growthbook
# GB-18 Support experiment CRUD events in event webhooks

### Features and Changes

The following changes add support for event webhook logging for Experiment-related CRUD events, sending the `ApiExperiment` payload as logged by events.

- It adds experiment CRUD actions to the UI so we can select the events
- It sends payloads for experiment CRUD actions via webhooks

It also changes the logging from `FeatureInterface` to `ApiFeature`. This changes what is in the events collection, removing the requirement to transform the event payload before sending it via webhooks, allowing us to remove all code related to transforming data for webhooks.


### Dependencies

n/a

### Testing

1. Create an event webhook for `feature.updated`, `feature.created`, `feature.deleted`, `experiment.updated`, `experiment.created`, `experiment.deleted`.
2. Perform actions to create, update and delete both features and experiments.
3. See that your event webhooks get sent. You can use the `webhooks-impl` project in the [examples repository](https://github.com/growthbook/examples)


### Screenshots

![CleanShot 2023-03-09 at 14 37 38@2x](https://user-images.githubusercontent.com/113377031/224176205-afee7085-105d-4b25-9f0e-5717c93e8a12.png)

<img width="939" alt="image" src="https://user-images.githubusercontent.com/113377031/224176277-aca1f456-e00e-45e3-94c1-04ef6b4d9b60.png">

Events received in the webhooks-impl project:

<img width="1163" alt="image" src="https://user-images.githubusercontent.com/113377031/224176435-b3bd5e3b-c5dd-496d-995f-aee9e2cf0a85.png">
